include/dlaf/solver/triangular/mc.tpp

```diff
@@ -93,18 +98,21 @@ void Solver<Backend::MC>::triangular(comm::CommunicatorGrid grid, blas::Side sid
                                      blas::Op op, blas::Diag diag, T alpha,
                                      Matrix<const T, Device::CPU>& mat_a,
                                      Matrix<T, Device::CPU>& mat_b) {
-  // Check if matrix A is square
-  DLAF_ASSERT_SIZE_SQUARE(mat_a);
-  // Check if block matrix A is square
-  DLAF_ASSERT_BLOCKSIZE_SQUARE(mat_a);
-  // Check compatibility of the communicator grid and the distribution of matrix A
-  DLAF_ASSERT_DISTRIBUTED_ON_GRID(grid, mat_a);
-  // Check compatibility of the communicator grid and the distribution of matrix B
-  DLAF_ASSERT_DISTRIBUTED_ON_GRID(grid, mat_b);
+  DLAF_ASSERT(matrix::square_size(mat_a), "Matrix is not square!", mat_a);
+  DLAF_ASSERT(matrix::square_blocksize(mat_a), "Matrix blocksize is not square!", mat_a);
+  DLAF_ASSERT(matrix::equal_process_grid(mat_a, grid),
+              "The matrix is not distributed according to the communicator grid!", mat_a);
```

Comment:
Ii would be a good idea to output also grid info, but I don't know if it is printable...

Reply:
It is not yet, but we can easily make it printable if there's consensus.